rbac/internal/views.py

```diff
@@ -504,6 +505,51 @@ class SentryDiagnosticError(Exception):
     pass
 
 
+def migration_resources(request):
+    """View or delete specific resources related to migration.
+    DELETE /_private/api/utils/migration_resources/?resource=xxx&org_id=xxx
+    GET /_private/api/utils/migration_resources/?resource=xxx&org_id=xxx&limit=1000
+    optinos of resource: workspace, mapping(tenantmapping), binding(bindingmapping)
+    """
+    resource = request.GET.get("resource")
+    if not resource:
+        return HttpResponse(
+            'Invalid request, must supply the "resource" query parameter.',
+            status=400,
+        )
+    resource = resource.lower()
+    resource_model = RESOURCE_MODEL_MAPPING.get(resource)
+    if not resource_model:
+        return HttpResponse(
+            f"Invalid request, resource should be in '{RESOURCE_MODEL_MAPPING.keys()}'.",
+            status=400,
+        )
+    
+    org_id = request.GET.get("org_id")
+
+    if request.method == "DELETE":
+        if not destructive_ok("api"):
+            return HttpResponse("Destructive operations disallowed.", status=400)
+        run_migration_resource_deletion.delay({"resource": resource, "org_id": org_id})
+        logger.info(f"Deleting resources of type {resource}. Requested by '{request.user.username}'")
+        return HttpResponse("Resource deletion is running in a background worker.", status=202)
+
+    elif request.method == "GET":
+        resource_objs = resource_model.objects.all()
+        if org_id:
+            tenant = get_object_or_404(Tenant, org_id=org_id)
+            if tenant.tenant_name == "public":
+                return HttpResponse("Invalid request, tenant cannot be the public one.", status=400)
+            if resource == "binding":
+                resource_objs = resource_objs.filter(role__tenant=tenant)
+            else:
+                resource_objs = resource_objs.filter(tenant=tenant)
```

Comment:
Maybe we can factor this org_id queryset filtering to a common place for both reads and deletes?

Reply:
Updated